alica_engine/include/engine/AlicaContext.h

```diff
@@ -0,0 +1,237 @@
+/**
+ * @file
+ * This file contains api interface of alica for applications.
+ */
+
+#pragma once
+
+#include <assert.h>
+#include <memory>
+#include <string>
+#include <type_traits>
+
+#include "engine/IBehaviourCreator.h"
+#include "engine/IConditionCreator.h"
+#include "engine/IConstraintCreator.h"
+#include "engine/IUtilityCreator.h"
+#include "engine/constraintmodul/ISolver.h"
+
+namespace alica
+{
+
+class AlicaEngine;
+class IAlicaCommunication;
+
+/**
+ * Alica options that can be set at runtime.
+ * They are either not part of conf files or app might want to override them at runtime.
+ */
+// TODO: shift them to separate header?
+#define ALICA_OPT_THREADPOOL_SIZE (0)
```

Comment:
enum class even

Reply:
I thought maybe in some cases we can pass option number and value as it is to other modules say ROS or posix sockets. Doesn't look like we have that case though.

Reply:
i don't think so, plus enum classes can always be converted to the base integer type if need be.
However, since you introduced a thread pool size parameter, it should also be implemented...